src/Space.ts

```diff
@@ -80,9 +102,16 @@ class SpaceService extends Base {
     const { apiEndpointAddress, got } = this.lensPlatformClient;
     const url = `${apiEndpointAddress}/spaces/${spaceName}`;
 
-    const json = await got.patch(url, {
-      json: space
-    });
+    const json = await throwExpected(
+      () => got.patch(url, {
```

Comment:
What if we use switch and case for this?

```
try {
    const json = got.patch(url, { json: space });
catch (error) {
    // assuming .status_code here
    switch (error.status_code) {
        case 500:
        throw new TokenNotFoundException();

        // ...

       default:
       throw error;
    }
}
```

Reply:
That's the problem: while debugging API errors in the extension I noticed that status code is often not easily accessible - that's why I had to write a separate function that attempts to extract it. Unless I'm missing something.

The mapping could definitely be refactored to simple closure with switch-case logic, but that would result in a requirement to provide default implementation for unexpected errors - which right now is done automatically in `throwExpected()`

Reply:
Makes sense